tests/components/recorder/test_statistics_v23_migration.py

```diff
@@ -176,42 +175,42 @@ def test_delete_duplicates(
                 schema_version_postfix=SCHEMA_VERSION_POSTFIX,
             ),
         ),
-        get_test_home_assistant() as hass,
     ):
-        recorder_helper.async_initialize_recorder(hass)
-        setup_component(hass, "recorder", {"recorder": {"db_url": recorder_db_url}})
-        get_instance(hass).recorder_and_worker_thread_ids.add(threading.get_ident())
-        wait_recording_done(hass)
-        wait_recording_done(hass)
-
-        with session_scope(hass=hass) as session:
-            session.add(
-                recorder.db_schema.StatisticsMeta.from_meta(external_energy_metadata_1)
-            )
-            session.add(
-                recorder.db_schema.StatisticsMeta.from_meta(external_energy_metadata_2)
-            )
-            session.add(
-                recorder.db_schema.StatisticsMeta.from_meta(external_co2_metadata)
-            )
-        with session_scope(hass=hass) as session:
-            for stat in external_energy_statistics_1:
-                session.add(recorder.db_schema.Statistics.from_stats(1, stat))
-            for stat in external_energy_statistics_2:
-                session.add(recorder.db_schema.Statistics.from_stats(2, stat))
-            for stat in external_co2_statistics:
-                session.add(recorder.db_schema.Statistics.from_stats(3, stat))
-
-        hass.stop()
+        async with async_test_home_assistant() as hass, async_test_recorder(hass):
+            get_instance(hass).recorder_and_worker_thread_ids.add(threading.get_ident())
```

Comment:
Would be nicer if we did this in the database executor, but its an existing issue

Reply:
Yeah, we can sort that out in a separate PR